packages/data-stores/src/utils/wpcom-wrapper.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import wpcomProxyRequest from 'wpcom-proxy-request';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'data-stores:utils:wpcom-wrapper' );
+
+export interface WpcomRequestParams {
+	path: string;
+	method?: string;
+	apiVersion?: string;
+	formData?: object;
+}
+
+export function wpcomRequest< T >( params: WpcomRequestParams ): Promise< T > {
+	return new Promise( ( resolve, reject ) => {
+		wpcomProxyRequest( params, ( err: object, res: T ) => {
```

Comment:
I'm more asking for advice here as I'm not sure: would it make sense to add type definitions to `wpcomProxyRequest` in the global declarations? 

That way the parser will complain if `wpcomProxyRequest` doesn't receive what it expects.

For example, in `packages/data-stores/global.d.ts`:

```
declare module 'wpcom-proxy-request' {
   // import only works if we add "baseUrl": "." to tsconfig.json compilerOptions
    import { WpcomRequestParams } from 'src/utils/wpcom-wrapper'; 
    export default function wpcomProxyRequest( params: WpcomRequestParams, callback: Function ): XMLHttpRequest;
}
```

Reply:
Oh, I like that idea. I'll have a play with that, too.

Reply:
I've added this in and it _seems_ to be working well so far, but I'm sure someone with more TS knowledge might have a better idea!